src/utils/array_utils.ts

```diff
@@ -24,6 +24,19 @@ export function insertWithoutDuplicates<T>(array: T[], item: T, newIndex: number
     return newArray;
 }
 
+export function insertMultipleWithoutDuplicates<T>(array: T[], items: T[], newIndex: number) {
+    let newArray = [...array];
```

Comment:
This does return a new array even if nothing moved, but that's probably fine. I can't imagine people will use the bulk move within a category very often. I can't think of a concise way to check if nothing moved anyway without just doing a shallow equals check.